pyro/infer/traceenum_elbo.py

```diff
@@ -50,6 +50,19 @@ def _check_model_guide_enumeration_constraint(model_enum_sites, guide_trace):
                                      .format(name, f.name))
 
 
+def _check_tmc_elbo_constraint(model_trace, guide_trace):
```

Comment:
`TraceEnum_ELBO` does not support taking multiple samples from nonreparameterized distributions in the model - it does not compute the corresponding DiCE factors and thus produces incorrect gradients.  This function checks for this case and conservatively raises a warning if any variables in the model are multiply sampled, even if they are reparameterized.